### ISSUE TITLE
Do we want to implement axe-matchers ruby gem to integrate accessibility into Rspec?

### ISSUE THREAD

**Issue by CGillen** [OregonDigital/OD2#535]
### Descriptive summary

https://github.com/dequelabs/axe-matchers axe-matchers provides a simple integration between Rspec and aXe automated accessibility testing. This will allow us to automate our own accessibility testing, give us a moving base line to not regress behind, and allow or force us to block pull requests that does regress us.

### Expected behavior

Choice whether or not to integrate axe-matchers and time allocation to integrate.

### Accessibility Concerns

Automates a baseline accessibility compliance. Does not substitute full manual audits and diligent defensive coding. May give us a false sense of security if we're not constantly aware.

**Comment by revgum** [comment]
I think it's a great idea and seems like it should fit right into our `system` tests that use the javascript driver and such..

**Comment by Nerdmaster** [comment]
This is worth using, I think.  But we need to be very careful about relying on this as anything more than a minor crutch in the war against inaccessible content.  It's automated checks that only run against the pages we define in our system tests - as those are the only tests which actually render content to a headless browser.  This means there are going to be a *lot* of gaps.

**Comment by raybrarian** [comment]
Agreed, lots of gaps! But I think any place we can add some considered support (especially coming from the folks at Deque) to help fill the automatable gaps, as long as we are explicit about what it **isn't** doing, can help us. I think Corey's characterization of that potential problem you are also flagging is accurate - that we need to proactively avoid that "false sense of security."

**Comment by jechols** [comment]
(From a prior comment I put in with the wrong account)

This is worth using, I think. But we need to be very careful about relying on this as anything more than a minor crutch in the war against inaccessible content. It's automated checks that only run against the pages we define in our system tests - as those are the only tests which actually render content to a headless browser. This means there are going to be a lot of gaps.

**Comment by CGillen** [comment]
Example output from work show page:

```
3.1) Failure/Error: expect(page).to be_accessible

            Found 4 accessibility violations:

            1) aria-allowed-role: ARIA role must be appropriate for the element (minor)
                https://dequeuniversity.com/rules/axe/3.3/aria-allowed-role?application=axeAPI
                The following 1 node violate this rule:

                    Selector: .breadcrumb
                    HTML: <nav class="breadcrumb" role="region" aria-label="Breadcrumb"><ol><li><a href="/?locale=en">Home</a></li><li><a href="/dashboard?locale=en">Dashboard</a></li><li><a href="/dashboard/my/works?locale=en">Works</a></li><li class="active">Test Title</li></ol></nav>
                    Fix any of the following:
                    - ARIA role region  is not allowed for given element

            2) color-contrast: Elements must have sufficient color contrast (serious)
                https://dequeuniversity.com/rules/axe/3.3/color-contrast?application=axeAPI
                The following 1 node violate this rule:

                    Selector: .label-success
                    HTML: <span class="label label-success">Public</span>
                    Fix any of the following:
                    - Element has insufficient color contrast of 2.48 (foreground color: #ffffff, background color: #5cb85c, font size: 11.0pt, font weight: bold). Expected contrast ratio of 4.5:1

            3) landmark-unique: Ensures landmarks are unique (moderate)
                https://dequeuniversity.com/rules/axe/3.3/landmark-unique?application=axeAPI
                The following 1 node violate this rule:

                    Selector: #masthead
                    HTML: <nav id="masthead" class="navbar navbar-static-top" role="navigation">
                    Fix any of the following:
                    - The landmark must have a unique aria-label, aria-labelledby, or title to make landmarks distinguishable

            4) link-name: Links must have discernible text (serious)
                https://dequeuniversity.com/rules/axe/3.3/link-name?application=axeAPI
                The following 4 nodes violate this rule:

                    Selector: a[title="Facebook"]
                    HTML: <a class="resp-sharing-button__link" target="_blank" rel="noopener noreferrer" title="Facebook" href="https://facebook.com/sharer/sharer.php?u=http%3A%2F%2Ftest%3A3001%2Fconcern%2Fgenerics%2F37720c723%3Flocale%3Den">
                    Fix all of the following:
                    - Element is in tab order and does not have accessible text
                    Fix any of the following:
                    - Element does not have text that is visible to screen readers
                    - aria-label attribute does not exist or is empty
                    - aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
                    - Element's default semantics were not overridden with role="presentation"
                    - Element's default semantics were not overridden with role="none"

                    Selector: a[title="Twitter"]
                    HTML: <a class="resp-sharing-button__link" target="_blank" rel="noopener noreferrer" title="Twitter" href="https://twitter.com/intent/tweet/?text=Generic+%7C+Test+Title+%7C+ID%3A+37720c723+%7C+Hyrax&amp;url=http%3A%2F%2Ftest%3A3001%2Fconcern%2Fgenerics%2F37720c723%3Flocale%3Den">
                    Fix all of the following:
                    - Element is in tab order and does not have accessible text
                    Fix any of the following:
                    - Element does not have text that is visible to screen readers
                    - aria-label attribute does not exist or is empty
                    - aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
                    - Element's default semantics were not overridden with role="presentation"
                    - Element's default semantics were not overridden with role="none"

                    Selector: a[title="Google\+"]
                    HTML: <a class="resp-sharing-button__link" target="_blank" rel="noopener noreferrer" title="Google+" href="https://plus.google.com/share?url=http%3A%2F%2Ftest%3A3001%2Fconcern%2Fgenerics%2F37720c723%3Flocale%3Den">
                    Fix all of the following:
                    - Element is in tab order and does not have accessible text
                    Fix any of the following:
                    - Element does not have text that is visible to screen readers
                    - aria-label attribute does not exist or is empty
                    - aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
                    - Element's default semantics were not overridden with role="presentation"
                    - Element's default semantics were not overridden with role="none"

                    Selector: a[title="Tumblr"]
                    HTML: <a class="resp-sharing-button__link" target="_blank" rel="noopener noreferrer" title="Tumblr" href="https://www.tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Ftest%3A3001%2Fconcern%2Fgenerics%2F37720c723%3Flocale%3Den&amp;posttype=link&amp;shareSource=tumblr_share_button">
                    Fix all of the following:
                    - Element is in tab order and does not have accessible text
                    Fix any of the following:
                    - Element does not have text that is visible to screen readers
                    - aria-label attribute does not exist or is empty
                    - aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
                    - Element's default semantics were not overridden with role="presentation"
                    - Element's default semantics were not overridden with role="none"

            Invocation: axe.run(callback);
```
